pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -0,0 +1,788 @@
+				}
+			}
+
+			// Now we need information from drivers.
+			podScheduling, err := state.initializePodScheduling(ctx, pod, pl.podSchedulingLister)
+			if err != nil {
+				return statusError(logger, err)
+			}
+			status := statusForClaim(podScheduling, pod.Spec.ResourceClaims[index].Name)
+			if status != nil {
+				for _, unsuitableNode := range status.UnsuitableNodes {
+					if node.Name == unsuitableNode {
+						return statusUnschedulable(logger, "resourceclaim cannot be allocated for the node (unsuitable)", "pod", klog.KObj(pod), "node", klog.KObj(node), "resourceclaim", klog.KObj(claim), "unsuitablenodes", status.UnsuitableNodes)
+					}
+				}
+			}
+		default:
+			// This should have been delayed allocation. Immediate
+			// allocation was already checked for in PreFilter.
+			return statusError(logger, fmt.Errorf("internal error, unexpected allocation mode %v", claim.Spec.AllocationMode))
+		}
+	}
+
+	if len(unavailableClaims) > 0 {
+		state.mutex.Lock()
+		defer state.mutex.Unlock()
+		if state.unavailableClaims == nil {
+			state.unavailableClaims = sets.NewInt()
+		}
+		state.unavailableClaims.Insert(unavailableClaims...)
+		return statusUnschedulable(logger, "resourceclaim not available on the node", "pod", klog.KObj(pod))
+	}
+
+	return nil
+}
+
+// PostFilter checks whether freeing an allocated claim might help to get a Pod
+// schedulable. This only gets called when filtering found no suitable node.
+func (pl *dynamicResources) PostFilter(ctx context.Context, cs *framework.CycleState, pod *v1.Pod, filteredNodeStatusMap framework.NodeToStatusMap) (*framework.PostFilterResult, *framework.Status) {
+	if !pl.enabled {
+		return nil, framework.NewStatus(framework.UnschedulableAndUnresolvable, "plugin disabled")
+	}
+	logger := klog.FromContext(ctx)
+	state, err := getStateData(cs)
+	if err != nil {
+		return nil, statusError(logger, err)
+	}
+	if len(state.claims) == 0 {
+		return nil, framework.NewStatus(framework.UnschedulableAndUnresolvable, "nothing done, no claims")
+	}
+
+	// Iterating over a map is random. This is intentional here, we want to
+	// pick one claim randomly because there is no better heuristic.
+	for index := range state.unavailableClaims {
+		claim := state.claims[index]
+		if len(claim.Status.ReservedFor) == 0 ||
+			len(claim.Status.ReservedFor) == 1 && claim.Status.ReservedFor[0].UID == pod.UID {
+			claim := state.claims[index].DeepCopy()
+			claim.Status.DeallocationRequested = true
+			claim.Status.ReservedFor = nil
+			logger.V(5).Info("Requesting deallocation of ResourceClaim", "pod", klog.KObj(pod), "resourceclaim", klog.KObj(claim))
+			if err := state.updateClaimStatus(ctx, pl.clientset, index, claim); err != nil {
+				return nil, statusError(logger, err)
+			}
+			return nil, nil
+		}
+	}
+	return nil, framework.NewStatus(framework.UnschedulableAndUnresolvable, "still not schedulable")
```

Comment:
+1 to change to reason, it's meaningless.

Reply:
So the suggested change is to return `framework.Unschedulable` here, right?

Reply:
For beta, let's consider whether it makes sense to make a distinction in the framework. For now, let's keep it as Unschedulable, as it's semantically more accurate (preemption will run).

This is the first time we have a 2nd PostFilter plugin, so we are not sure how to go from here.

Reply:
I was about to say the same and already started writing an issue about it so that we don't forget :grin: 

https://github.com/kubernetes/kubernetes/issues/113854

Reply:
But when you say " keep it as Unschedulable", do you mean keep the current code with `framework.UnschedulableAndUnresolvable` or switch to `framework.Unschedulable`?

Reply:
I think `framework.Unschedulable` makes more sense, I just haven't switched to it yet.

Reply:
I just pushed an update with `framework.Unschedulable`.